Add Self-assessment Docs (#78)
kubernetes/sig-security: sig-security-assessments/Documentation/recipe-book.md

to have you here. Below is an outline of the journey to get to a successful Self-assessment.

## Preparation
Set yourself up for success – build your team and get organized
1.	Figure out who can do this Self-assessment with you. You will need a security expert and a counterpart who owns the 
project and is an expert in it. If you are requesting a Security Self-assessment you can reach out to the team in
[#sig-security](https://kubernetes.slack.com/messages/sig-security) to see if there is someone interested in helping to
champion the assessment.
1.	Make sure that you and your counterpart each have a back-up person assigned who attends all the meetings and is 
involved in the assessment. Two heads are better than one, and redundancy means that if someone can’t make a meeting 
for whatever reason, that you can still make progress.
1.	To get further participation, advertise in SIG meetings and channels for the project, as well as SIG-Security. Aim for a minimum of 4 
people (a project expert, a security expert, and a backup for each), and a maximum of 8. This ensures that there is 
not overwhelming burden on any one individual but allows for a smooth decision-making process.

> 1.	Create a Google doc for meeting notes using 

Author: question: At what point do we want Self-assessment project teams to open a request for a slack channel?

Reviewer: Hm - I think once there is a solid group of people who are interested in participating in the self assessment that a slack channel becomes necessary for coordination and publicity/socializing that a self-assessment is underway. /2p

# Recipe Book 👩‍🍳
🎉Yay! You have decided to do a Self-assessment and make your corner of Kubernetes more secure! We are so happy 
to have you here. Below is an outline of the journey to get to a successful Self-assessment.

## Preparation
Set yourself up for success – build your team and get organized

> 1.	Figure out who can do this Self-assessment with you. You will need a security expert and a counterpart who owns the 

Author: If we are allowing this to be a bit more self serve should we have a dedicated sub-project channel for folks to easily find potential security champions for their assessments? 

We could also have an intake form so project owners could make a formal request to start a Self-assessment?

> 1.	Open a Security Assessment Request using the [issue template.](https://github.com/kubernetes/sig-security/issues/new/choose)

Reviewer 2: (nit!)
```suggestion
1.	Open a Security Assessment Request using the [issue template](https://github.com/kubernetes/sig-security/issues/new/choose).
```

project and is an expert in it. If you are requesting a Security Self-assessment you can reach out to the team in
[#sig-security](https://kubernetes.slack.com/messages/sig-security) to see if there is someone interested in helping to

> champion the assessment.

Author: If we are allowing this to be a bit more self serve should we have a dedicated sub-project channel for folks to easily find potential security champions for their assessments? 

We could also have an intake form so project owners could make a formal request to start a Self-assessment?

Reviewer: There is a dedicated Self Assessment channel already that I think we can totally use to source security champions: https://kubernetes.slack.com/archives/C0441E11REC. Do you think that's sufficient?

The intake form lives here when you add an issue to sig-security: https://github.com/kubernetes/sig-security/issues/new/choose

1.	Make sure that you and your counterpart each have a back-up person assigned who attends all the meetings and is 
involved in the assessment. Two heads are better than one, and redundancy means that if someone can’t make a meeting 
for whatever reason, that you can still make progress.

> 1.	To get further participation, advertise in other SIG meetings. Aim for a minimum of 4 people (a project expert, a 

Author: @aladewberry - is this intended for someone in the process of starting a Self-assessment (requester) or is this someone looking for projects that may want a Self-Assessment?

> 1.	To get further participation, advertise in SIG meetings and channels for the project, as well as SIG-Security. Aim for a minimum of 4 

Reviewer 2: (nit)
```suggestion
1.	To get further participation, advertise in SIG meetings and channels for the project, as well as SIG Security. Aim for a minimum of 4 
```

people (a project expert, a security champion, and a backup for each), and a maximum of 8. This ensures that there is 
not overwhelming burden on any one individual but allows for a smooth decision-making process.
1. Once you have a dedicated group of individuals ready to start the Security Assessment you can open an issue to request

> a slack channel for the project specific assessment - [example here.](https://github.com/kubernetes/community/pull/7015)

Reviewer 2: (nit)
```suggestion
a Slack channel for the project specific assessment - here's an [example](https://github.com/kubernetes/community/pull/7015).
```

> 1.	Create a Google doc for meeting notes using 

Reviewer 2: ```suggestion
1.	Create a Google doc for meeting notes using the
```

[Kubernetes community meeeting notes template](https://github.com/kubernetes/community/blob/master/events/community-meeting.md) 
## Meetings
- Once you have your collateral and your team together, set up a kickoff meeting with everyone. It is likely that not 
everyone will show up. That’s ok!
- Keep meeting notes using the doc you created during the preparation phase.

> - Create a recurring meeting series of about ~6 meetings (this is flexible and can be extended if necessary).

Reviewer 2: (nit)
```suggestion
- Create a recurring meeting series of about 6 meetings (this is flexible and can be extended if necessary).
```

- These sessions can be ad hoc and should be set for whatever timing works for the group.
## The Assessment
### Step One
Draw a [data flow diagram](https://www.lucidchart.com/pages/data-flow-diagram) for the main workflows in the project. 

> This step will show you the areas you can potentially assess, and allow the security expert to quickly gain context 

Author: I'm flip flopping between the term "security champion" and "security expert" - does anyone have strong opinions one way or another?

Reviewer: I think Champion is more inviting than Expert. One of the principles here that I want the subproject to espouse is that you don't have to be an expert to do something to positively augment your security posture. Having *some* security knowledge and acting on it is a good thing, in other words. Yes, we absolutely need to partner with experts to make sure that things are properly thought through, thorough, etc. But any non-expert should feel empowered to start a self-assessment and to reach out to experts for guidance, which is what we're here for.

around how the project is structured - and where the biggest security concerns may be. 
[Excalidraw](https://excalidraw.com/) is a great tool for this!
### Step Two 
Decide what to assess - defining the scope of the assessment is essential to making it successful. This decision will
keep you focused and ensure you make forward progress. Helpful questions when thinking about what to assess:
- What workflows are used most?
- What workflows does the community want a security assessment of?
- What expertise does the team doing the assessment have? Does it match any of the potential flows? 
Try to get matching expertise for the scope!
### Step Three 
Now that you have scope defined, we can come up with threats that are reasonable and logical for the projects.

> - Use the [TAG Self-assessment template](https://github.com/cncf/tag-security/blob/main/assessments/guide/self-assessment.md) 

Author: Are all fields relevant and useful in the TAG-Security Self-Assessment template or would it be useful to create a fork of their template for this project?

Reviewer 3: Fork would be nice especially after we have lessons learnt from vsphere csi-driver assessment too

Author: I'll bring this up in the next meeting.

Author: I will not do that as part of this PR - but as a follow up. We will fork the TAG-Security one, and then we can iterate on it and make it more suited to our use cases.

Reviewer: @cailynse @PushkarJ  just wanted to check in - is there any action I need to take here?

Author: For the template I'll open a separate PR once this one is merged.

For this PR we need a /lgtm from you and then an approval from a chair I believe!

to write up the report. You can use Google Docs or HackMD – whatever the project team is most comfortable with. 
- Once the team is happy with the above doc, convert it into a markdown PR and assign reviewers.

> - Now is a good time to start writing a [blog post](https://github.com/kubernetes/website/tree/main/content/en/blog) to 

Reviewer 2: ```suggestion
- Now is a good time to start writing a [blog post](https://kubernetes.io/docs/contribute/new-content/blogs-case-studies/) to
```

project and is an expert in it. If you are requesting a Security Self-assessment you can reach out to the team in
[#sig-security](https://kubernetes.slack.com/messages/sig-security) to see if there is someone interested in helping to
champion the assessment.
1.	Make sure that you and your counterpart each have a back-up person assigned who attends all the meetings and is 
involved in the assessment. Two heads are better than one, and redundancy means that if someone can’t make a meeting 
for whatever reason, that you can still make progress.
1.	To get further participation, advertise in other SIG meetings. Aim for a minimum of 4 people (a project expert, a 
security expert, and a backup for each), and a maximum of 8. This ensures that there is not overwhelming burden on any
one individual but allows for a smooth decision-making process.
1.	Create a Google doc for meeting notes using 
[Kubernetes community meeeting notes template](https://github.com/kubernetes/community/blob/master/events/community-meeting.md) 
for future and past meetings for tracking. Share these notes in 
[#sig-security](https://kubernetes.slack.com/messages/sig-security), 

> [#tag-security](https://cloud-native.slack.com/archives/CDJ7MLT8S), and the appropriate Kubernetes Slack channel for 

Author: question: Do we want to do this? Might we just share the assessment slack channel, and maybe an invite to the meeting to the projects slack channel since they may not be on our mailing list?

Reviewer 3: IMO, Probably okay to skip.
